### PR TITLE
[SC-1219] upgrade to new fee flow

### PR DESCRIPTION
**Pull request by ZumZoom**

### CODE REVIEW

File: contracts/SimpleSettlement.sol [hunk 1]
```diff
@@ -3,42 +3,173 @@
...
         bytes calldata extraData
-    ) internal virtual override(ResolverValidationExtension, IntegratorFeeExtension) {
-        super._postInteraction(order, extension, orderHash, taker, makingAmount, takingAmount, remainingMakingAmount, extraData);
+    ) internal view override returns (uint256) {
+        (uint256 rateBump, bytes calldata tail) = _getRateBump(extraData);
+        return Math.mulDiv(
+            super._getMakingAmount(order, extension, orderHash, taker, takingAmount, remainingMakingAmount, tail),
+            _BASE_POINTS,
+            _BASE_POINTS + rateBump
+        );
```
Comment (zZoMROT) [inline contracts/SimpleSettlement.sol:48]:
```suggestion
            _BASE_POINTS + rateBump,
            Math.Rounding.Floor
        );
```

File: contracts/SimpleSettlement.sol [hunk 2]
```diff
@@ -3,42 +3,173 @@
...
+        bytes calldata extension,
+        bytes32 orderHash,
+        address taker,
+        uint256 makingAmount,
+        uint256 remainingMakingAmount,
+        bytes calldata extraData
+    ) internal view override returns (uint256) {
+        (uint256 rateBump, bytes calldata tail) = _getRateBump(extraData);
+        return Math.mulDiv(
+            super._getTakingAmount(order, extension, orderHash, taker, makingAmount, remainingMakingAmount, tail),
```
Comment (zZoMROT) [inline contracts/SimpleSettlement.sol:65]: Looks like we should get fee from the result takingAmount, not intermediate calculations - at first auction calculations, then getting fee calculation

File: contracts/SimpleSettlement.sol [hunk 3]
```diff
@@ -3,42 +3,173 @@
...
+                }
+                currentRateBump = nextRateBump;
+                currentPointTime = nextPointTime;
+                pointsAndTimeDeltas = pointsAndTimeDeltas[5:];
+            }
+            return ((auctionFinishTime - block.timestamp) * currentRateBump / (auctionFinishTime - currentPointTime), tail);
+        }
+    }
+
+    /**
```
Comment (zZoMROT) [inline contracts/SimpleSettlement.sol:144]: Wrong natspec's content

File: contracts/SimpleSettlement.sol (hunk 1 above)
Comment (ZumZoom) [reply contracts/SimpleSettlement.sol:48]: This just add additional computations without any benefits see [here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/0df841d2d75352786532e9e26f400efdef78d86e/contracts/utils/math/Math.sol#L229). `mulDiv` without rounding parameter uses floor rounding by default.

File: contracts/SimpleSettlement.sol (hunk 3 above)
Comment (ZumZoom) [reply contracts/SimpleSettlement.sol:144]: fixed in 82f0a25

**Review by zZoMROT — APPROVED** [review]

File: test/MeasureGas.js [hunk 4]
```diff
@@ -20,37 +20,32 @@ describe('MeasureGas', function () {
         const [owner, alice] = await ethers.getSigners();
         const chainId = await getChainId();
         const abiCoder = ethers.AbiCoder.defaultAbiCoder();
-        const { dai, weth, inch, accessToken, lopv3, lopv4 } = await deploySwapTokens();
+        const { dai, weth, accessToken, lopv3, lopv4 } = await deploySwapTokens();
 
         await dai.transfer(alice, ether('100'));
-        await inch.mint(owner, ether('100'));
+        // await inch.mint(owner, ether('100'));
```
Comment (byshape) [inline test/MeasureGas.js:26]:
```suggestion
```

**Pull request merged**